ui/package.json

```diff
@@ -12,53 +12,58 @@
     "test": "jest",
     "lint": "eslint --cache --format codeframe --ext mjs,jsx,js src test"
   },
+  "comment": "IMPORTANT! date-fns is pinned to 1.3.x due to https://github.com/mui-org/material-ui-pickers/issues/1440",
+  "comment2": "IMPORTANT! react-virtualized is held at 9.21.1 due to https://github.com/bvaughn/react-virtualized/issues/1226 breaking scrolling to anchors on ListRules",
+  "comment3": "see also https://github.com/bvaughn/react-virtualized/issues/1179#issuecomment-572854217",
+  "comment4": "reverting this package probably won't fix anything",
   "dependencies": {
-    "@date-io/date-fns": "^1.3.7",
-    "@hot-loader/react-dom": "^16.8.6",
-    "@material-ui/core": "^4.2.1",
-    "@material-ui/icons": "^4.2.1",
-    "@material-ui/lab": "4.0.0-alpha.20",
-    "@material-ui/pickers": "^3.1.2",
-    "@material-ui/styles": "^4.2.1",
-    "@mozilla-frontend-infra/components": "^2.4.1",
-    "axios": "^0.18.1",
-    "change-case": "^3.1.0",
+    "@date-io/date-fns": "^1.3.13",
+    "@hot-loader/react-dom": "^16.11.0",
+    "@material-ui/core": "^4.8.3",
+    "@material-ui/icons": "^4.5.1",
+    "@material-ui/lab": "4.0.0-alpha.39",
+    "@material-ui/pickers": "^3.2.8",
+    "@material-ui/styles": "^4.8.2",
+    "@mozilla-frontend-infra/components": "^3.0.4",
+    "axios": "^0.19.1",
+    "change-case": "^4.1.1",
     "classnames": "^2.2.6",
-    "clipboard-copy": "^3.0.0",
-    "codemirror": "^5.48.4",
-    "date-fns": "2.0.0-alpha.37",
+    "clipboard-copy": "^3.1.0",
+    "codemirror": "^5.50.2",
+    "date-fns": "2.9.0",
     "deep-sort-object": "^1.0.2",
     "diff": "^4.0.1",
-    "downshift": "^3.2.10",
-    "is-absolute-url": "^2.1.0",
-    "mdi-react": "^5.5.0",
+    "downshift": "^4.0.4",
+    "is-absolute-url": "^3.0.3",
+    "mdi-react": "^6.5.0",
     "prop-types": "^15",
-    "qs": "^6.7.0",
+    "qs": "^6.9.1",
     "ramda": "^0.26.1",
-    "react": "^16.8.6",
+    "react": "^16.12.0",
     "react-auth0-components": "^1.0.3",
     "react-codemirror2": "^6.0.0",
-    "react-diff-view": "^2.1.4",
-    "react-dom": "^16.8.6",
+    "react-diff-view": "^2.4.1",
+    "react-dom": "^16.12.0",
     "react-helmet": "^5.2.1",
-    "react-hot-loader": "^4",
-    "react-number-format": "^4.0.8",
-    "react-router-dom": "^5.0.1",
-    "react-virtualized": "^9.21.1",
-    "unidiff": "^1.0.2"
+    "react-hot-loader": "^4.12.18",
+    "react-number-format": "^4.3.1",
+    "react-router-dom": "^5.1.2",
+    "react-virtualized": "^9.21.2",
+    "unidiff": "^1.0.2",
+    "upper-case": "^2.0.1"
```

Comment:
Hmm, maybe we can just not install upper-case. There's a built-in [`.toUpperCase`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toUpperCase) method that we can use.

Reply:
I'll double check and do this if possible.